### PR TITLE
WIP ref #94: Use Symfony Flex

### PR DESCRIPTION
**Pull request by umulmrum**
- Adjust to new directory structure
- Use new Symfony dependencies
- Make use of both environment and parameters.yaml

| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | #94 
| License       | MIT

Corresponding PR to https://github.com/chameleon-system/chameleon-base/pull/92 in chameleon-base (see that PR for further notices).

When using Flex, there is normally no more parameters.yaml. I decided to keep it for now, though, as we have a rather large number of parameters and I do not like to force all projects to move to environment variables at once. Instead we should push back parameter usage in chameleon-base and convert more parameters to bundle config. Then the developers can decide on PROJECT level whatever configuration type should be used.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# This file defines all environment variables that the application needs.
+# DO NOT DEFINE PRODUCTION SECRETS IN THIS FILE.
+# Use ".env.local" for local overrides during development.
+# Use real environment variables when deploying to production.
+# https://symfony.com/doc/current/best_practices/configuration.html#infrastructure-related-configuration
+
+###> symfony/framework-bundle ###
+APP_ENV=dev
+APP_SECRET=CHANGEME
```
Comment (UlrichKu) [inline .env]: This isn't a .env.dist?

File: config/config.inc.php [hunk 2]
```diff
@@ -3,18 +3,6 @@
...
-    if ('console' === REQUEST_PROTOCOL) {
-        $input = new \Symfony\Component\Console\Input\ArgvInput();
-        $env = $input->getParameterOption(array('--env', '-e'), getenv('SYMFONY_ENV') ?: 'dev');
-        $devMode = 'prod' !== $env;
-    } else {
-        if ('.intra' === substr($_SERVER['HTTP_HOST'], -6)) {
-            $devMode = true;
-        }
-    }
-    define('_DEVELOPMENT_MODE', $devMode);
```
Comment (UlrichKu) [inline config/config.inc.php]: What does now distinguish between runtime modes?

File: config/parameters.yaml.dist [hunk 3]
```diff
@@ -1,25 +1,17 @@
...
-    secret: !ThisTokenIsNotSoSecretChangeIt!
-    chameleon_system_core.debug.show_view_source_html_hints: true
-    chameleon_system_core.development_email: "the developer's mail address"
-
-    database_host: "localhost"
-    database_port: 3306
-    database_name: "shp_dmo61"
-    database_user: "shp_dmo61"
-    database_password: "insert-password"
+    chameleon_system_core.development_email: "the developer's email address"
```
Comment (UlrichKu) [inline config/parameters.yaml.dist]: Is this still necessary / or newly used?

File: .gitignore [hunk 4]
```diff
@@ -1,32 +1,37 @@
...
+/bin/
+/composer.phar
+/public/assets/less/chameleon.css
+/public/chameleon/mediapool/*
+!/public/chameleon/mediapool/.gitkeep
 
-web/chameleon/outbox/*
-!web/chameleon/outbox/static/
-web/chameleon/outbox/static/*
+/public/chameleon/outbox/*
```
Comment (UlrichKu) [inline .gitignore]: This renaming of web to public -> doesn't this need more code changes as shown in this PR?

Or are all users of the "public directory" using already a (correctly set) constant instead?

File: .gitignore [hunk 5]
```diff
@@ -1,32 +1,37 @@
...
+!/public/chameleon/outbox/static/
```
Comment (UlrichKu) [inline .gitignore]: And this rename does influence external services: ie where an apache would look for relevant filest?

File: config/config.inc.php (hunk 2 above)
Comment (bestform) [reply config/config.inc.php]: The `APP_ENV` environment variable.

File: .env (hunk 1 above)
Comment (umulmrum) [reply .env]: .env handling best practice was changed by Symfony just last week: https://symfony.com/blog/improvements-to-the-handling-of-env-files-for-all-symfony-versions

File: config/parameters.yaml.dist (hunk 3 above)
Comment (umulmrum) [reply config/parameters.yaml.dist]: It's unchanged and still required.

File: .gitignore (hunk 4 above)
Comment (umulmrum) [reply .gitignore]: We use either PATH_WEB or $_SERVER['DOCUMENT_ROOT'], so there should be no problems (ironically we switched from public to web a few years ago, so we already know that we're "clean").

Nevertheless there will be references to the web directory in comments, docs and database-stored help texts which I didn't think of yet --> should be searched and replaced.

File: .gitignore (hunk 5 above)
Comment (umulmrum) [reply .gitignore]: Yes, this will need to be addressed in docs.

**Pull request closed without merging**